### DISCUSSION TITLE
cva@1.0 (beta)

### DISCUSSION THREAD

**Discussion by joe-bell** [joe-bell/cva#205] [Announcements]
After many months of development, ~~`class-variance-authority@1.0`~~ `cva@1.0` is now in **`beta`**!

**Visit [beta.cva.style](https://beta.cva.style/getting-started/whats-new) and share your thoughts below**

In the meantime, I'll be cracking through a long list of things to do ✌️

**Comment by flexbox** [comment]
Congrats @joe-bell 🙌

**Comment by Siregar** [comment]
Thanks for your hard work. It is greatly appreciated and I look forward to the stable 1.0 version. 🙏

**Comment by magoz** [comment]
Amazing work @joe-bell! 🔥
[The option to use Tailwind Merge with cva](https://beta.cva.style/getting-started/installation/#handling-style-conflicts) is a huge DX improvement!

**Comment by relliv** [comment]
Congratulations. I'm looking forward to it 👍

**Comment by belgattitude** [comment]
Fantastic work ! Thanks a lot. ~Out of curiosity will it be published under the "cva" name ? Wondering cause I saw a package exists on npm https://www.npmjs.com/package/cva~ **edit** the install doc answered my questions :)

**Comment by alexandrm-opti** [comment]
I am very happy to use you package. 
Any plan to releases 1.0.0 stable version?

**Reply by flexbox** [reply]
Of course 😉 

1. Go to this page https://joebell.co.uk/sponsors
2. Send 145 ETH to the wallet

**Reply by joe-bell** [reply]
It will happen, but no ETA currently 

Personal life comes first!

**Reply by lu-zen** [reply]
You know, I'm afraid this becomes too big and complex that you need to abandon it. keep it simple pls 😆

**Reply by joe-bell** [reply]
Not planning to add too much; maybe 1-2 more features at most

After that, I'll probably consider this library to be feature complete for the foreseeable future

**Comment by RIP21** [comment]
I see in the list of improvements minification.
What it means? There is simply less code or that you minify your code before shipping to NPM? If it's about minfication through some uglifier then, I may be wrong, but I think this is considered not a good practice. AFAIK, just shipping code compiled down to, let's say, Node 16 compatible syntax is how it's supposed to be and let the consumer bundle/minify it however they like for their production.

Here are more points with which I agree: https://www.briefs.fm/3-minutes-with-kent/221

Maybe I'm missing something. And if I do, will be glad to be educated :) Also, I see you just use SWC to build the package. Highly recommend using https://github.com/egoist/tsup to simplify your building setup slightly bit :)

**Reply by joe-bell** [reply]
Completely forgot to reply, but I reverted the minification changes after you shared this – thanks for the heads up!

**Comment by nyambe** [comment]
Great news,  I just discovered this post.  I am using the original cva with vue3/nuxt3 and tailwind  Let me know if I can be of any help.

**Comment by 0xbuidler** [comment]
Looking forward the beta becoming stable 🙏

**Comment by jeniabrook** [comment]
@joe-bell huge fan of your work with `cva`!

Do you still plan to release `cva@1.0`?

**Reply by joe-bell** [reply]
Yep!

**Reply by jeniabrook** [reply]
@joe-bell I Just saw your response—awesome work!  

If you need any help with it, I’d be more than happy to assist!

**Comment by jeroengerits** [comment]
Thanks @joe-bell for all the great work!

**Comment by ehellman** [comment]
@joe-bell do you accept contributions for v1? :) the lack of composition in 0.x is a huge pain and for sure a killer feature in v1. going to be using cva in the project I am in for 1-1.5 years at least so I'd love to help push through a stable v1 if you need an extra pair of hands

**Reply by joe-bell** [reply]
Really appreciate the gesture @ehellman! Unfortunately, managing over contributors also takes up quite a bit of time to co-ordinate

At the moment, personal life takes priority, but shipping `cva@1.0` is something I'm determined to do

**Comment by brrock** [comment]
> Thanks for your hard work. It is greatly appreciated and I look forward to the stable 1.0 version. 🙏

@shadcn

**Comment by nyambe** [comment]
Good job, will checkout

**Reply by shaun-fitzgibbon** [reply]
What ever happened to this ? As all theise posts are atleast a year old and no version 1 is out ? Is it still in active development ?

**Reply by joe-bell** [reply]
No change in answer since my previous comment: this is a project undertaken in my free-time outside of my full-time job. Still want to ship, but my time is finite and personal life takes priority

**Comment by codayon** [comment]
Nice!

**Comment by shirshendubhowmick** [comment]
Hello,

Thanks for all the work.

Any ETA on when can we expect a stable v1.0.0 ?

**Reply by joe-bell** [reply]
Hey! No ETA but chipping away in my spare time
